pennylane/transforms/decompositions/utils.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions required for decomposing two-qubit operations."""
+
+from pennylane import math
+from pennylane import numpy as np
+
+# This gate E is called the "magic basis". It can be used to convert between
+# SO(4) and SU(2) x SU(2). For A in SO(4), E A E^\dag is in SU(2) x SU(2).
+E = np.array([[1, 1j, 0, 0], [0, 0, 1j, 1], [0, 0, 1j, -1], [1, -1j, 0, 0]]) / np.sqrt(2)
+Edag = E.conj().T
+
+LAST_COL_NEG = np.diag([1, 1, 1, -1])  # used to negate the last column of a matrix
+
+
+def _convert_to_su4(U):
```

Comment:
it feels a bit odd to have private functions in a _different_ module to where they are used 🤔 

I think it makes more sense to make them public now that they are separate